inference-engine/samples/benchmark_app/main.cpp

```diff
@@ -378,18 +378,21 @@ int main(int argc, char *argv[]) {
             topology_name = cnnNetwork.getName();
             slog::info << (FLAGS_b != 0 ? "Network batch size was changed to: " : "Network batch size: ") << batchSize << slog::endl;
 
-            // ----------------- 6. Configuring input ----------------------------------------------------------------------
+            // ----------------- 6. Configuring inputs and outputs ----------------------------------------------------------------------
             next_step();
 
-            for (auto& item : inputInfo) {
-                if (app_inputs_info.at(item.first).isImage()) {
-                    /** Set the precision of input data provided by the user, should be called before load of the network to the device **/
+            processPrecision(cnnNetwork, FLAGS_ip, FLAGS_op, FLAGS_iop);
+            for (auto& item : cnnNetwork.getInputsInfo()) {
+                // if precision for input set by user, then set it to app_inputs
+                // if it an image, set U8
+                if (!FLAGS_i.empty() || FLAGS_iop.find(item.first) != std::string::npos) {
```

Comment:
`if (!FLAGS_ip.empty() || FLAGS_iop.find(item.first) != std::string::npos) {`

Reply:
OMG. Indeed. Thanks for catching that.